### PR TITLE
Fixed jpeg-turbo diagnostics and build options in default configuration.

### PR DESCRIPTION
**Pull request by asmorkalov**
`cmake /opencv`.

Before:
```
--   Media I/O: 
--     ZLib:                        /lib64/libz.so (ver 1.2.7)
--     JPEG:                        libjpeg-turbo (ver 2.1.3-62)
--     WEBP:                        build (ver encoder: 0x020f)
--     PNG:                         /usr/local/lib/libpng.so (ver 1.6.40)
--     TIFF:                        build (ver 42 - 4.6.0)
--     JPEG 2000:                   build (ver 2.5.0)
--     OpenEXR:                     build (ver 2.3.0)
```

After:
```
--   Media I/O: 
--     ZLib:                        /lib64/libz.so (ver 1.2.7)
--     JPEG:                        build-libjpeg-turbo (ver 2.1.3-62)
--       SIMD Support Request:      YES
--       SIMD Support:              YES
--     WEBP:                        build (ver encoder: 0x020f)
--     PNG:                         /usr/local/lib/libpng.so (ver 1.6.40)
--     TIFF:                        build (ver 42 - 4.6.0)
--     JPEG 2000:                   build (ver 2.5.0)
--     OpenEXR:                     build (ver 2.3.0)
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by vpisarev** [comment]
@asmorkalov, can you maybe provide imread() performance numbers before and after your patch? to make sure that the change fixes the problem

**Comment by asmorkalov** [comment]
Yes, I'm working on it right now.

**Comment by asmorkalov** [comment]
```
Geometric mean (ms)

Name of Test  4.x-1  patched-1 patched-1
                                   vs    
                                 4.x-1   
                               (x-factor)
Decode::JPEG 70.120   36.749      1.91   
Encode::JPEG 111.216  28.581      3.89   
decode::PNG  31.451   30.872      1.02   
encode::PNG  58.192   57.151      1.02
```

File: 3rdparty/libjpeg-turbo/CMakeLists.txt [hunk 1]
```diff
@@ -56,8 +56,7 @@ if(MSVC_IDE AND CMAKE_GENERATOR_PLATFORM MATCHES "arm64")
   set(CPU_TYPE arm64)
 endif()
 
-OCV_OPTION(ENABLE_LIBJPEG_TURBO_SIMD "Include SIMD extensions for libjpeg-turbo, if available for this platform" (NOT CV_DISABLE_OPTIMIZATION)
-  VISIBLE_IF BUILD_JPEG)
```
Comment (mshabunin) [inline 3rdparty/libjpeg-turbo/CMakeLists.txt:60]: Maybe we should add `BUILD_JPEG_TURBO` variable instead and check it here? Currently logic is not very obvious, i.e. in order to disable own JPEG library users must provide `BUILD_JPEG=OFF` **and** `BUILD_JPEG_TURBO_DISABLE=ON`.

**Comment by asmorkalov** [comment]
The PR has been discussed on the Core team meeting. Decided to merge as is without modifications for now to presume backward compatibility with existing scripts. Also BUILD_XXX variables are not set automatically for JPEG and some other 3rdparties. It's need to be fixed with dedicated PR.

**Pull request merged**